File: mambo-protocol/src/main/scala/org/mambo/protocol/types/game/context/fight/FightTeamMemberCharacterInformations.scala





















// Generated on 05/03/2013 10:35:04
// Scala version by Blackrush
package org.mambo.protocol.types

import org.mambo.core.io._
import org.mambo.protocol.{Type => BaseType, _}
import org.mambo.protocol.enums._

trait FightTeamMemberCharacterInformations extends FightTeamMemberInformations {
  override val typeId = 13

  protected trait Type extends FightTeamMemberInformations.Type with Serializable {
    override val typeId = FightTeamMemberCharacterInformations.typeId

    val name: String
    val level: Short

    override def serialize(buf: Buffer) {
      super.serialize(buf)

      buf.writeString(name)
      buf.writeShort(level)
    }

  }

  protected def create(id: Int, name: String, level: Short): Target = throw new IllegalAccessException

  override def deserialize(buf: Buffer) = {
    val id = buf.readInt()
    val name = buf.readString()
    val level = buf.readShort()

    create(id, name, level)
  }

}

object FightTeamMemberCharacterInformations extends FightTeamMemberCharacterInformations {
  case class TypeImpl(id: Int, name: String, level: Short) extends Type
  override def create(id: Int, name: String, level: Short) = TypeImpl(id, name, level)

  def unapply(msg: TypeImpl) = TypeImpl.unapply(msg)
  def apply(id: Int, name: String, level: Short) = create(id, name, level)

  type Target = TypeImpl

}

File: mambo-protocol/src/main/java/org/mambo/protocol/types/FightTeamMemberCharacterInformations.java


// Generated on 05/08/2013 19:38:00
package org.mambo.protocol.types;

import java.util.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class FightTeamMemberCharacterInformations extends FightTeamMemberInformations {
    public static final short TYPE_ID = 13;
    
    @Override
    public short getTypeId() {
        return TYPE_ID;
    }
    
    public String name;
    public short level;
    
    public FightTeamMemberCharacterInformations() { }
    
    public FightTeamMemberCharacterInformations(int id, String name, short level) {
        super(id);
        this.name = name;
        this.level = level;
    }
    
    @Override
    public void serialize(Buffer buf) {
        super.serialize(buf);
        buf.writeString(name);
        buf.writeShort(level);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        super.deserialize(buf);
        name = buf.readString();
        level = buf.readShort();
        if (level < 0)
            throw new RuntimeException("Forbidden value on level = " + level + ", it doesn't respect the following condition : level < 0");
    }
    
}
